bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -85,6 +85,29 @@ def run_plan(scope, plan_name, named_args = {})
       executor.report_yaml_plan(closure.model.body)
     end
 
+    # If a TargetSpec parameter is passed, ensure it is in inventory
+    inventory = Puppet.lookup(:bolt_inventory)
+    # Is it possible that inventory is not here?
+    if inventory.version > 1
+      parameter_names_to_type = closure.parameters.each_with_object({}) do |param, param_acc|
+        # parameter has multiple keys ex Optional[TargetSpec]
+        param_acc[param.name] = if defined?(param.type_expr.keys)
+                                  param.type_expr.keys.each_with_object([]) do |param_type, type_acc|
+                                    type_acc << param_type.cased_value
+                                  end
+                                # Only a single key, may or may not have a type specified
+                                else
+                                  param.type_expr ? [param.type_expr.cased_value] : []
+                                end
+      end
+
+      params.each do |param, value|
+        if parameter_names_to_type[param].include?('TargetSpec')
```

Comment:
Does this handle aliases?

Reply:
Ah, good question. When I was writing tests (pushed after your initial review) for this in `boltlib` I realized I needed `BoltLib::TargetSpec` so updated it to match `/TargetSpec/` and left a `CODEREVIEW` question. I think I for sure need look in to this more.

Reply:
Updated to handle type aliases